hips/hip-0019.md

```diff
@@ -0,0 +1,61 @@
+
+## Rationale
+The aim of this proposal is to grant developers and integrators more control over the linting process by enabling:
+- Exclusions for files imported or managed by third parties that are not directly editable.
+- Exclusions for files that may not adhere to static linting rules.
+- Exclusions for files containing intentional deviations from standard practices due to specific deployment scenarios.
+
+The `.helmlintconfig.yaml` file will support simple patterns for matching files and directories, enhancing ease of management. The `--lint-config-file` flag will facilitate centralized management of lint exclusions in complex projects.
+
+## Specification
+### `.helmlintconfig.yaml` File Format
+The `.helmlintconfig.yaml` file enables chart developers to specify filenames or glob patterns to exclude from linting. The format is YAML, with patterns specified under a `lintIgnore` key, making future extensions for additional configurations feasible.
+
+#### Example
+```
+# .helmlintconfig.yaml file example
+lintIgnore:
+  - fileMask: "migrations/templates/job.yaml"                                       # Ignores all errors from this file
+  - errorMask: "chart metadata is missing these dependencies**"                     # Ignores all icon errors
+  - fileMask: "gitlab/templates/shared-secrets/self-signed-cert-job.yml"
+    errorMask: "<include 'fullname' .>"                                             # Ignores this specific error but not other errors in the file
+```
+
+### Command Behavior
+When `helm lint` is executed, it checks for the presence of a `.helmlintconfig.yaml` file in the chart directory or at a location specified by the `--lint-config-file` flag. The command then parses this file to exclude any files or directories matching the patterns specified, including specific error lines.
+
+## Backwards Compatibility
+This proposal introduces no breaking changes. Charts without a `.helmlintconfig.yaml` file or those not using the `--lint-config-file` flag will function as before.
+
+## Security Implications
+No significant security implications are expected as the `.helmlintconfig.yaml` file is processed locally by `helm lint`, without modifying chart content or involving network activity.
+
+## How to Teach This
+Documentation for the `.helmlintconfig.yaml` file and the `--lint-config-file` flag will be integrated into the official Helm documentation under the `helm lint` section. Detailed examples and common patterns will be provided to assist new users.
+
+## Reference Implementation
+PR #[13257](https://github.com/helm/helm/pull/13257) created in the Helm/helm repository in gitlab 
+
+## Open Issues
+- Developing an effective and user-friendly pattern syntax for the `.helmlintconfig.yaml` file.
+- Considering the potential for in-file annotations to temporarily exclude specific chart sections from linting.
+
+## Rejected Ideas
```

Comment:
Providing rule specific excludes and configurations, and composing configuration from "default" configuration e.g. similar to https://yamllint.readthedocs.io/en/stable/configuration.html#extending-the-default-configuration

Helm today doesn't "name" its lint rules, and changing the code/structure to support the above can be extended later

Reply:
Done, let me know if it's ok

Reply:
oops, sorry, I mean we should add these to the "rejected ideas":
- Providing rule specific excludes (Helm today doesn't "name" its lint rules)
- composing configuration from "default" configuration (e.g. similar to https://yamllint.readthedocs.io/en/stable/configuration.html#extending-the-default-configuration)